synapse/events/utils.py

```diff
@@ -679,7 +679,7 @@ def validate_canonicaljson(value: Any) -> None:
     * Floats
     * NaN, Infinity, -Infinity
     """
-    if isinstance(value, int):
+    if type(value) is int:
```

Comment:
I think this is pretty much a no-op, but good to do since we do call out `bool` below.

Reply:
Yeah. Before this would catch `bool`s here and do nothing. Now they get caught at the bottom.

Reply:
Just to clarify -- they *still* do nothing. This isn't a behavior change.